week-9/Homework/mandatory/1-practice/2-code-reading.md

```diff
@@ -61,3 +74,12 @@ console.log(y);
 ```
 
 What will be the output of this code. Explain your answer in 50 words or less.
+
+## Answer 3
+
+9
+{x:9}
+
+When we call f1 function with the argument of x variable, it will return 10,
+but this doesn't change global x and console on next line will log 9.
+Same logic should be with declaration oy y object globally, and changing its key/value inside function scope, it won't change global y(not sure about this).
```

Comment:
For this one, it's actually going to change
The reason is how JS functions work. 
To get more info, search Passing by value vs passing by references 
in a nutshell, all primitive type of variables are pass by value type, but objects will pass by reference 
https://hackernoon.com/grasp-by-value-and-by-reference-in-javascript-7ed75efa1293